Disallow Works to be shown if they are not published

rschenk opened the pull request:
When rendering a Work, the resource page requires/assumes that there will be at least one published version. This was causing trouble when a user _with edit access_ requested the resource page for a draft work, which caused a 500. This commit tweaks the policy to raise a pundit error when requesting the resource page of draft work.

Note that #717 was only ever an issue when the user requesting said work had edit access to it.

Closes #717

cam156 commented on app/services/field_configurator.rb:
```diff
@@ -9,29 +9,36 @@ def self.common_fields
+             date_created: FieldConfig.new("Date Created"),
+             rights: FieldConfig.new("Rights"),
+             identifier: FieldConfig.new("Identifier"),
+             description: FieldConfig.new("Description"))
```
@awead isn't this going to put the description at the bottom of the page?

awead replied:
@cam156 no, in fact, AFAICT, this doesn't affect anything relating to the show view because that's all done with the _attribute_rows.html.erb partial, and description is a rendered separately in the _work_description partial.

rschenk committed:
Disallow Works to be shown if they are not published

When rendering a Work, the resource page requires/assumes that there
will be at least one published version. This was causing trouble when a
user _with edit access_ requested the resource page for a draft work,
which caused a 500. This commit tweaks the policy to raise a pundit
error when requesting the resource page of draft work.

awead commented on app/policies/work_policy.rb:
```diff
@@ -8,7 +8,7 @@ def limit
 
   def show?
-    (record.discover_access?(user) && published?) || edit?
+    published? && (record.discover_access?(user) || edit?)
```
@rschenk If I'm reading this right, nobody (not even admins) can view unpublished _works_. Work _versions_ that are in draft state should still be fine, but I'm wondering if there's a need to redirect to the draft work version if an admin, or someone else, attempts to view the unpublished work uuid.

I may be over-thinking this, because in most cases, we aren't going to be passing around uuids for unpublished works, just for work versions since that's the url in the dashboard.

rschenk replied:
You're reading that right. I'm not sure if this is the way we want to go, but it does seem that way. The resource page for a _Work_ requires that it have at least one published version. If there is not a published version it will panic and 500. 

We could either make the resource page gracefully handle unpublished works, or we disallow them entirely (what this does).

rschenk committed:
Remove dead Work History page
